modules/apps/segments/segments-service/src/main/java/com/liferay/segments/service/impl/SegmentsExperimentLocalServiceImpl.java

```diff
@@ -462,6 +464,55 @@ private DynamicQuery _getSegmentsExperienceIdsDynamicQuery(
 		return dynamicQuery;
 	}
 
+	private void _publishSegmentsExperienceVariant(
+		SegmentsExperience controlSegmentsExperience,
+		SegmentsExperience variantSegmentsExperience) {
+
+		int lowestSegmentsExperiencePriority = Optional.ofNullable(
+			segmentsExperiencePersistence.fetchByG_C_C_Last(
+				controlSegmentsExperience.getGroupId(),
+				controlSegmentsExperience.getClassNameId(),
+				controlSegmentsExperience.getClassPK(), null)
+		).map(
+			SegmentsExperience::getPriority
+		).orElse(
+			SegmentsExperienceConstants.PRIORITY_DEFAULT
+		);
+
+		int controlSegmentsExperiencePriority =
+			controlSegmentsExperience.getPriority();
+		int variantSegmentsExperiencePriority =
+			variantSegmentsExperience.getPriority();
+
+		controlSegmentsExperience.setPriority(
+			lowestSegmentsExperiencePriority - 1);
+
+		controlSegmentsExperience = segmentsExperiencePersistence.update(
+			controlSegmentsExperience);
+
```

Comment:
Maybe if you don't flush the value you can obtain a constraint error here?

Reply:
I see your point, but It's not happening https://github.com/liferay/liferay-portal/blob/master/modules/apps/segments/segments-service/src/main/java/com/liferay/segments/service/impl/SegmentsExperienceLocalServiceImpl.java#L569 either

Reply:
But it is not happening because we don't call the swap per every case https://github.com/liferay/liferay-portal/blob/6137d1f86886b79cb92d02a541fd8cdf1720a181/modules/apps/segments/segments-service/src/main/java/com/liferay/segments/service/impl/SegmentsExperienceLocalServiceImpl.java#L447

Reply:
neither here

https://github.com/liferay/liferay-portal/blob/11e6081f96abb6bf299369519434c1eafa0658e3/modules/apps/segments/segments-service/src/main/java/com/liferay/segments/service/impl/SegmentsExperimentLocalServiceImpl.java#L522

Reply:
Maybe it is happening here too, we don't have enough tests of this method to discard there is a bug in this line either I will create a test, before say this piece of code is working, since we are finding so many bugs in this part of the code.

Reply:
When you're right, you're right. Yes, I'm going to add test for every "problematic" use case. Thanks!!